server/persistence/iwclient/iwclient_list_test.go

```diff
@@ -197,21 +211,18 @@ var _ = Describe("List", func() {
 
 		It("should be returned in list of owner's workspaces", func() {
 			// when the list of workspaces owned by 'owner-user' is requested
-			var ww workspacesv1alpha1.InternalWorkspaceList
-			err := c.ListAsUser(ctx, "owner-user", &ww)
+			var iww workspacesv1alpha1.InternalWorkspaceList
+			err := c.ListAsUser(ctx, "owner-user", &iww)
 			Expect(err).NotTo(HaveOccurred())
 
 			// then the list contains just the 'owner-ws' workspace
-			wwi := ww.Items
-			Expect(wwi).Should(HaveLen(len(wwi)))
-
-			for _, w := range wwi {
-				sw := slices.ContainsFunc(wwi, func(z workspacesv1alpha1.InternalWorkspace) bool {
-					zll, wll := z.GetLabels(), w.GetLabels()
-					return zll[workspacesv1alpha1.LabelDisplayName] == wll[workspacesv1alpha1.LabelDisplayName] &&
-						zll[workspacesv1alpha1.LabelWorkspaceOwner] == wll[workspacesv1alpha1.LabelWorkspaceOwner]
+			Expect(iww.Items).To(HaveLen(len(ww)))
+
+			for _, w := range iww.Items {
+				sw := slices.ContainsFunc(ww, func(z *workspacesv1alpha1.InternalWorkspace) bool {
+					return z.Spec.DisplayName == w.Spec.DisplayName && z.Spec.Owner.JwtInfo.Sub == w.Spec.Owner.JwtInfo.Sub
 				})
-				Expect(sw).Should(BeTrue())
+				Expect(sw).To(BeTrue())
```

Comment:
Rather than use `slices.ContainsFunc`, we could use the [`Satisfy`] inside a [`ContainElement`] matcher.  Alternatively, two [`HaveField`] matchers wrapped in [`And`] would also work.  Might look a bit like this:

<details><summary>Diff</summary>
<p>

```diff
diff --git a/server/persistence/iwclient/iwclient_list_test.go b/server/persistence/iwclient/iwclient_list_test.go
index a216077..2df4bea 100644
--- a/server/persistence/iwclient/iwclient_list_test.go
+++ b/server/persistence/iwclient/iwclient_list_test.go
@@ -3,7 +3,6 @@ package iwclient_test
 import (
 	"context"
 	"fmt"
-	"slices"
 
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
@@ -219,10 +218,10 @@ var _ = Describe("List", func() {
 			Expect(iww.Items).To(HaveLen(len(ww)))
 
 			for _, w := range iww.Items {
-				sw := slices.ContainsFunc(ww, func(z *workspacesv1alpha1.InternalWorkspace) bool {
-					return z.Spec.DisplayName == w.Spec.DisplayName && z.Spec.Owner.JwtInfo.Sub == w.Spec.Owner.JwtInfo.Sub
-				})
-				Expect(sw).To(BeTrue())
+				Expect(ww).To(ContainElement(
+					Satisfy(func(z *workspacesv1alpha1.InternalWorkspace) bool {
+						return z.Spec.DisplayName == w.Spec.DisplayName && z.Spec.Owner.JwtInfo.Sub == w.Spec.Owner.JwtInfo.Sub
+					})))
 			}
 		})
 

```

</p>
</details> 

[`And`]: https://pkg.go.dev/github.com/onsi/gomega@v1.33.1#And
[`Satisfy`]: https://pkg.go.dev/github.com/onsi/gomega@v1.33.1#Satisfy
[`ContainElement`]: https://pkg.go.dev/github.com/onsi/gomega@v1.33.1#ContainElement
[`HaveField`]: https://pkg.go.dev/github.com/onsi/gomega@v1.33.1#HaveField

Reply:
very nice, let me apply this. Thx!